### DISCUSSION TITLE
Problem passing extra args using routing with a specific app

### DISCUSSION THREAD

**Discussion by GDuroS** [anvilistas/anvil-extras#486] [Q&A] [answered]
In one of my apps I have a search page with filters and a repeating panel to show the results. Each item in the panel has a button to navigate the user to the edit page of the item selected.

I navigate the following way:
`routing.set_url_hash(f"filmes/edit/{self.item['Sequence']}", filme_row=self.item)`
In the edit page, the "constructor" is:
`def __init__(self, filme_row=None, **properties):`
And this works great! Inside the init method I check the id and query the row only if it was not already passed from the previous page.

However, I'm doing the same in another app and everything that I pass as extra arguments is always passed as the default value (None).

Any ideias of what could it be?

**Comment by s-cork** [comment]
Sounds odd. Think I'd need a minimal example clone link. Particularly if it's working in one situation but not another.

**Reply by GDuroS** [reply]
Yeah... It works everytime in an app, but never in the other. It doesn't matter if I pass a row of a DataTable or a string, nothing works.

About the clone link... It's kind of hard, since my app depends on the user table and on a few dependencies of mine. I'll try to make it simple in a clone to send you, since I'm sure there must be something that I'm doing it wrong, even though I can't seem to find it.

**Reply by GDuroS** [reply]
Ok. I think I made it minimal...

https://anvil.works/build#clone:7V5VR2ME3LW3NMOX=4KTUBMSWGT25NDX4GJR4J5WT

If you run the app, it will load the Interfaces.Home form as a route inside the Routers.MainRouter template. The page will only have a component that has a text box and a green button. You can type "seven" and a space to make an option load below. Click on it and the green button will change. Click on the green button.

This will make a route to Interfaces.Locais.LocalCadastro route inside the Routers.EditRouter passing a Sequence and the data row. I made prints to show the value that is being passed and the value received.

That's pretty much it.

**Reply by s-cork** [reply]
Thanks. That's not very minimal though. If you can make an app that only demonstrates the misbehaviour I'd appreciate it.

**Reply by s-cork** [reply]
Ok think this should be fixed in #487

**Reply by GDuroS** [reply]
Sorry, I tried to make as minimal as possible, without changing much inside the example forms to "preserve" the issue, since I didn't know if they were the problem.

About the fix, I just tested and in the development version of the dependency already works, so thanks very much! Is there any time frame for the fix to be merged into published version?

**Reply by s-cork** [reply, marked as answer]
that's pushed as v2.5.4
